docs/paper/dev/event-api/custom-events.md

```diff
@@ -0,0 +1,122 @@
+---
+slug: /dev/event-api/custom-events
+---
+
+# Custom Events
+
+Creating custom events is a great way to add functionality to your plugin. 
+This will allow for other people to listen for your custom events and add functionality to your plugin.
+
+## Creating a custom event
+
+To create a custom event, you need to create a class that extends `Event`. Each event has a `HandlerList` that contains all the listeners that are listening for that event.
+
+This list is used to call the listeners when the event is called.
+    
+```java title="PaperIsCoolEvent.java"
+
+public class PaperIsCoolEvent extends Event {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    public static HandlerList getHandlerList() {
+        return HANDLER_LIST;
+    }
+
+    @Override
+    public HandlerList getHandlers() {
+        return HANDLER_LIST;
+    }
+}
+```
+
+Now that we have created our event, we can add some functionality to it. 
+Perhaps this will contain a message that will be broadcast to the server when the event is called.
+
+```java title="PaperIsCoolEvent.java"
+
+public class PaperIsCoolEvent extends Event {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+    private final Component message;
+
+    public PaperIsCoolEvent(Component message) {
+        this.message = message;
+    }
+
+    public static HandlerList getHandlerList() {
+        return HANDLER_LIST;
+    }
+
+    @Override
+    public HandlerList getHandlers() {
+        return HANDLER_LIST;
+    }
+
+    public Component getMessage() {
+        return this.message;
+    }
+}
+```
+
+## Calling the event
+
+Now that we have created our event, we can call it.
+
+```java title="ExamplePlugin.java"
+    
+public class ExamplePlugin extends JavaPlugin {
+
+    // ...
+
+    public void callCoolPaperEvent() {
+        PaperIsCoolEvent coolEvent = new PaperIsCoolEvent(Component.text("Paper is cool!"))
+        coolEvent.callEvent();
+        Bukkit.broadcast(coolEvent.getMessage());
```

Comment:
Even though it is a bit obvious, for clarity, I'd add a comment here saying when other plugins will receive/modify message.